apps/qubit/modules/digitalobject/actions/metadataComponent.class.php

```diff
@@ -36,10 +36,13 @@ public function execute($request)
             && arStorageServiceUtils::getAipDownloadEnabled()
         );
 
-        $this->masterCopy = $this->resource->getRepresentationByUsage(QubitTerm::MASTER_ID);
         $this->referenceCopy = $this->resource->getRepresentationByUsage(QubitTerm::REFERENCE_ID);
         $this->thumbnailCopy = $this->resource->getRepresentationByUsage(QubitTerm::THUMBNAIL_ID);
 
+        if (!isset($this->resource)) {
```

Comment:
I think this conditional block should be moved above line 29 because line 29 & 30 will throw an error if `$this->resource` is null.